### PR TITLE
refactor(monorepo): relocate superset-ui

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

This PR MUST wait for [monorepo-stage-2](https://github.com/apache/superset/pull/17437) to pass CI.
This PR MUST merged before [monorepo-stage-2](https://github.com/apache/superset/pull/17437) .

1. add `lerna.json` in `superset-frontend`
2. add `workspace` in packages.json
3. run command `npx lerna import --flatten  --dest temporary_superset_ui ../../superset-ui` to import all commits into superset-frontend. commit cut to : https://github.com/apache-superset/superset-ui/commit/3e1699542808f0720a6c05d95fc95f557d5c90fc
4. discard `workspace` changing in packages.json (in order not to add new dependencies, dependencies will be added in stage-2 PR)
5. ignore `temporary_superset_ui` in pre-commit-config
6. remove `*.ipynb` from .gitignore
7. in order to support merge or squash this PR so temporarily update .asf.yml

-------------------------------------
To ensure that every PR passes the CI, the monorepo-related PR is split into multiple stages. 
1. https://github.com/apache/superset/pull/17427
2. https://github.com/apache/superset/pull/17445 (this PR)
3. https://github.com/apache/superset/pull/17437

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by ktmud — APPROVED** [review]

**Review by ktmud — COMMENTED** [review]
Thanks for working on this! This is the really large undertaking.

Just a reminder that when merging this PR, DON'T choose "Squash and merge", otherwise you'll lose all the git history.

**Comment by ktmud** [comment]
Before merge this PR, we must update [.asf.yml](https://github.com/apache/superset/blob/master/.asf.yaml) to enable "Create merge commit", otherwise it is not possible:

Xnip2021-11-16_16-19-22

**Review by geido — APPROVED** [review]
LGTM!

**Review by villebro — APPROVED** [review]
LGTM, but let's not merge until non-squash merge is enabled

File: .asf.yaml
Comment (kgabryje) [inline .asf.yaml:53]: It probably needs to be changed and merged in a separate PR

Comment (zhaoyongjie) [reply .asf.yaml]: open a new [pr](https://github.com/apache/superset/pull/17560) to handle this.

**Review by kgabryje — APPROVED** [review]
LGTM

**Comment by yening-zy** [comment]
This is really a huge change, it's too friendly for developers, what version will be launched?

**Comment by villebro** [comment]
We are just about to make a new release cut for 1.5. Finalizing the first release candidate + shipping it usually takes some time, but you should be able to start testing the new release branch within the next few weeks

**Pull request merged**
